Updated DG and PPP (#238)
AY2122S2-CS2103T-W09-4/tp: docs/DeveloperGuide.md

      Use case resumes from step 2.

> **Use case: UC05 - Assign task a student**

Reviewer: ```suggestion
**Use case: UC05 - Assign a task to a student**
```

**MSS**

> 1. User requests TAPA to assign a task to a student, by specifying their studentId and task name or assign tasks to all the student taking a module, by specifying module code and task name.

Reviewer: ```suggestion
1. User requests TAPA to assign a task to a student. (This can be done for an individual student or for all students taking the same module)
```

If i remember correctly, use cases should not have too much implementation details so just a suggestion to remove some details.

2. TAPA shows the updated list of students.

    Use case ends.

**Extensions**

* 1a. If studentId is given and a student with the given studentId does not exist.

    * 1a1. TAPA shows an error message.

      Use case ends.

* 1b. If module code is given and no student is taking a module with the given module code.

    * 1b1. TAPA shows an error message.

      Use case ends.

> * 1c. The task name already is assigned to the present.

Reviewer: ```suggestion
* 1c. The student has already been assigned with a task that has the same name.
```

    * 1c1. TAPA shows an error message.

      Use case ends.

> **Use case: UC06 - delete task assigned to student**

Reviewer: ```suggestion
**Use case: UC06 - Delete a task assigned to a student**
```

**MSS**

1. User requests to list students.
2. TAPA shows a list of students.

> 3. User requests to delete a task that was assigned to a student by specifying their studentId and the index of the task or delete a task assigned to all students taking a certain module by specifying the module code and task name.

Reviewer: ```suggestion
3. User requests to delete a task that was assigned to a student. (This can be done for an individual student or for all the students taking the same module.)
```

Simlar to above, just a suggestion to remove some implementation details.

4. TAPA updates the details of the student with the supplied details.

   Use case ends.

**Extensions**

* 2a. The list is empty.

> 

Reviewer: If the list is empty, would an error message be shown?

Author: An error will not be shown if the list is empty, but the use case for this function would end as we cannot perform step 3 and step 4.

  Use case ends.

* 3a. If studentId is given and a student with the given studentId does not exist.

    * 3a1. TAPA shows an error message.

      Use case ends.

> * 3b. If the index is out of range, that is a non-positive number or greater than the number of tasks assigned to that person.

Reviewer: Not too sure if this would be too much details for the implementation.

Author: I'll update into something less technical, thanks for the suggestion 👍
